doc-surrealdb_versioned_docs/version-2.x/reference-guide/security-best-practices.mdx

```diff
@@ -0,0 +1,323 @@
+
+**Examples**
+
+```surql
+DEFINE USER username ON DATABASE PASSWORD 'CHANGE_THIS' DURATION FOR SESSION 5d;
+```
+
+```surql
+DEFINE ACCESS account ON DATABASE TYPE RECORD
+	SIGNUP ( CREATE user SET email = $email, pass = crypto::argon2::generate($pass) )
+	SIGNIN ( SELECT * FROM user WHERE email = $email AND crypto::argon2::compare(pass, $pass) )
+	DURATION FOR SESSION 12h
+;
+```
+
+Tokens, however, are usually stored in the client (e.g. a web browser) and may be stolen by client-side attacks such as a cross-site scripting vulnerability in your application. For this reason, we strongly recommend reducing token duration from the default one hour to the minimum amount of time that your use case can tolerate. Ideally, a token should only be valid for as log as the client needs in order to use the token to establish a session, which can be as little as a few seconds.
+
+**Examples**
+
+```surql
+DEFINE USER username ON DATABASE PASSWORD 'CHANGE_THIS' DURATION FOR TOKEN 15m;
+```
+
+```surql
+DEFINE ACCESS account ON DATABASE TYPE RECORD
+	SIGNUP ( CREATE user SET email = $email, pass = crypto::argon2::generate($pass) )
+	SIGNIN ( SELECT * FROM user WHERE email = $email AND crypto::argon2::compare(pass, $pass) )
+	DURATION FOR TOKEN 5s
+;
+```
+
+## Query Safety
+
+When using SurrealDB as a traditional backend database, your application will usually build SurrealQL queries that may need to contain some untrusted input, such as that provided by the users of your application. To do so, SurrealDB offers [`bind`](https://surrealdb.com/docs/sdk/rust/setup#query) as a method to `query` (implemented in other SDKs as [the `vars` argument to `query`](https://surrealdb.com/docs/sdk/javascript/setup#query)), which should always be used when including untrusted input into queries. Otherwise, SurrealDB will be unable to separate the actual query syntax from the user input, resulting in the well-known [SQL injection](https://en.wikipedia.org/w/index.php?title=SQL_injection&oldid=1234729055) vulnerabilities. This practice is known as [prepared statements or parametrised queries](https://en.wikipedia.org/w/index.php?title=Prepared_statement&oldid=1195122133). Binding parameters ensures that untrusted data is passed to SurrealDB as SurrealQL parameters, which are independent from the query syntax, preventing SQL injection attacks.
+
```

Comment:
```suggestion
When using SurrealDB as a traditional backend database, your application will usually build SurrealQL queries that may need to contain some untrusted input, such as that provided by the users of your application. To do so, SurrealDB offers [`bind`](/docs/sdk/rust/setup#query) as a method to `query` (implemented in other SDKs as [the `vars` argument to `query`](/docs/sdk/javascript/core/data-querying), which should always be used when including untrusted input into queries. Otherwise, SurrealDB will be unable to separate the actual query syntax from the user input, resulting in the well-known [SQL injection](https://en.wikipedia.org/wiki/SQL_injection) vulnerabilities. This practice is known as [prepared statements or parametrised queries](https://en.wikipedia.org/wiki/Prepared_statement). 

Binding parameters ensure that untrusted data is passed to SurrealDB as SurrealQL parameters, which are independent of the query syntax, preventing SQL injection attacks.
```

Reply:
I agree with the suggestion, but you reverted the changes that I made to Wikipedia links as requested by @Dhghomon :sweat_smile:

Reply:
ah my bad! I added the links back but kept the text formatting